packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -4,24 +4,26 @@
 
 import 'package:meta/meta.dart';
 
+import '../artifacts.dart';
 import '../base/common.dart';
 import '../base/file_system.dart';
 import '../base/logger.dart';
 import '../convert.dart';
 import '../flutter_manifest.dart';
+import '../globals.dart' as globals;
```

Comment:
We're trying to avoid introducing more usage of`globals` into the codebase. (it exists for to allow for a gradual migration).

Instead you could either accepting the artifact path/process manager as dependencies in `generateLocalizations` or in the method that ultimately calls format. Fixing this also means removing the change to `testUsingContext` which is similarly to be avoided

Reply:
> Instead you could either accepting the artifact path/process manager as dependencies in `generateLocalizations` or in the method that ultimately calls format.

Do we have any examples for this?

Reply:
BotDetector is a simple class that injects all of its own dependencies: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/base/bot_detector.dart

Reply:
Sorry, but I don't think I'm getting progress in passing artifacts here. Could you provide more detailed guidance?

Reply:
Thanks for contributing this PR. Unfortunately, offering technical guidance to contributors is outside the scope of our code review process (and simply would not scale to the number of PRs that we receive).